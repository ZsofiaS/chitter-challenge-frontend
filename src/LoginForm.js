import React, { Component } from 'react';

class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: '',
      password: ''
    }
  }
  nameChangeHandler = (e) => {
    this.setState({
      userName: e.target.value
    })
  }
  passwordChangeHandler = (e) => {
    this.setState({
      password: e.target.value
    })
  }

  render() {
    return (
      <>
        <input id='nameInput' type="text" placeholder="Your name..." onChange={this.nameChangeHandler} />
        <input id='passwordInput' type="password" placeholder="Your password..." onChange={this.passwordChangeHandler} />
        <button
          id='loginSubmit'
          className="button shortButton"
          onClick={(event) => this.props.loginHandler(event, this.state.userName, this.state.password)}>Login</button>
        <button
          id='signupSubmit'
          className="button shortButton"
          onClick={(event) => this.props.signupHandler(event, this.state.userName, this.state.password)}>Signup</button>
      </>
    )
  }
}

export default LoginForm;
