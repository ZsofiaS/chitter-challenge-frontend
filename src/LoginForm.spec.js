import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import LoginForm from './LoginForm';

describe('LoginForm', () => {
  let wrapper;

  test('snapshot renders', () => {
    const tree = renderer.create(<LoginForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  beforeEach(() => wrapper = shallow(<LoginForm />));

  it('should render login button', () => {
    expect(wrapper.find('button').length).toEqual(2);
  })
})
