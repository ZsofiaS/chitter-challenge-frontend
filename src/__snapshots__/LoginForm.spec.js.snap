// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot renders 1`] = `
Array [
  <input
    id="nameInput"
    onChange={[Function]}
    placeholder="Your name..."
    type="text"
    value=""
  />,
  <input
    id="passwordInput"
    onChange={[Function]}
    placeholder="Your password..."
    type="password"
    value=""
  />,
  <button
    className="button shortButton"
    id="loginSubmit"
    onClick={[Function]}
  >
    Login
  </button>,
  <button
    className="button shortButton"
    id="signupSubmit"
    onClick={[Function]}
  >
    Signup
  </button>,
]
`;
